### PR TITLE
US-3.3.5: Add the gamification total points and segmentation

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 1]
```diff
@@ -122,4 +122,30 @@ export function replyInvitationToConnect(relationId, reply) {
       throw new Error ('Error when replying invitation to connect');
 	}
   })
-}
\ No newline at end of file
+}
+
+export function getEarnedPoints() {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: getGamificationPoints

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 2]
```diff
@@ -122,4 +122,30 @@ export function replyInvitationToConnect(relationId, reply) {
...
+    if(resp && resp.ok) {
+    return resp.json();
+    }
+    else {
+    throw new Error ('Error when getting earned points');
+    }
+  })
+}
+
+export function getGamificationPointStats() {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: getGamificationPointsStats

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 3]
```diff
@@ -122,4 +122,30 @@ export function replyInvitationToConnect(relationId, reply) {
...
+
+export function getEarnedPoints() {
+  return fetch(`/portal/rest/gamification/api/v1/points?userId=${eXo.env.portal.userName}`, {
+    method: 'GET',
+  }).then((resp) => {
+    if(resp && resp.ok) {
+    return resp.json();
+    }
+    else {
+    throw new Error ('Error when getting earned points');
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Error when getting gamification points

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 4]
```diff
@@ -137,6 +138,7 @@
       this.getSpacesRequestsSize();
       this.getConnectionsSize();
       this.getConnectionsRequestsSize();
+      this.getEarnedPoints()
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: this.getGamificationPoints();

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 5]
```diff
@@ -175,6 +177,13 @@
           }
         )
       },
+      getEarnedPoints() {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: getGamificationPoints()

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 6]
```diff
@@ -117,7 +117,7 @@
   </v-flex>
 </template>
 <script>
-  import {getUserInformations, getSpaces, getSpacesRequests, getConnections, getConnectionsRequests} from '../profilStatsAPI'
+  import {getUserInformations, getSpaces, getSpacesRequests, getConnections, getConnectionsRequests, getEarnedPoints} from '../profilStatsAPI'
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: getGamificationPoints

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 7]
```diff
@@ -175,6 +177,13 @@
           }
         )
       },
+      getEarnedPoints() {
+        getEarnedPoints().then(
+          (data) => {
+            this.totalPoints = data.points
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: this.totalPoints = data.points;

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 8]
```diff
@@ -175,6 +177,13 @@
           }
         )
       },
+      getEarnedPoints() {
+        getEarnedPoints().then(
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: getGamificationPoints()

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 9]
```diff
@@ -79,22 +81,30 @@
...
     },
-      methods: {
-        toProfileStats() {
-          this.$emit('isProfileStats');
-        }
+    created() {
+      this.getGamificationPointStats()
+    },
+    methods: {
+      getGamificationPointStats() {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: getGamificationPointsStats()

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 10]
```diff
@@ -79,22 +81,30 @@
...
           ]
         }
       };
     },
-      methods: {
-        toProfileStats() {
-          this.$emit('isProfileStats');
-        }
+    created() {
+      this.getGamificationPointStats()
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: this.getGamificationPointsStats();

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 11]
```diff
@@ -79,22 +81,30 @@
...
+    },
+    methods: {
+      getGamificationPointStats() {
+        getGamificationPointStats().then(
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: getGamificationPointsStats()

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 12]
```diff
@@ -42,13 +42,15 @@
...
+  import {getGamificationPointStats} from '../profilStatsAPI'
+
   export default {
     components: {
       'v-chart': ECharts
     },
     data() {
-      return {         
+      return {  
+        totalPoints : 0,
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: totalPoints : '',

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 13]
```diff
@@ -79,22 +81,30 @@
...
+    created() {
+      this.getGamificationPointStats()
+    },
+    methods: {
+      getGamificationPointStats() {
+        getGamificationPointStats().then(
+          (data) => {
+            this.option.series[0].data = JSON.parse(JSON.stringify(data).split('"label":').join('"name":'));
+            for (let i = 0; i < data.length; i++) {
+              this.totalPoints += data[i].value;
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: I think that it should be better to get the total points like in UserDashbord.vue

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 14]
```diff
@@ -79,22 +81,30 @@
...
                   
               },
-              data:[
-                {value:335, name:'Development',itemStyle: {color: 'rgb(31, 119, 180)'},},
-                {value:310, name:'Feedback',itemStyle: {color: 'rgb(44, 160, 44)'},},
-                {value:234, name:'Knowledge',itemStyle: {color: 'rgb(214, 39, 40)'},},
-                {value:200, name:'Social',itemStyle: {color: 'rgb(148, 103, 189)'},},
-                {value:500, name:'Teamwork',itemStyle: {color: 'rgb(255, 127, 14)'},}
-              ],
+              data:[],
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Not needed I think

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js (hunk 1 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js (hunk 2 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js (hunk 3 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue (hunk 4 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue (hunk 5 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue (hunk 6 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue (hunk 7 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue (hunk 8 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 9 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 10 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 11 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 12 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 13 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: updated

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 14 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: You're right

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue [hunk 15]
```diff
@@ -79,22 +80,33 @@
...
         }
       };
     },
-      methods: {
-        toProfileStats() {
-          this.$emit('isProfileStats');
-        }
+    created() {
+      this.getGamificationPointsStats();
+      this.getGamificationPoints()
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Add ; here

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 16]
```diff
@@ -137,6 +138,7 @@
       this.getSpacesRequestsSize();
       this.getConnectionsSize();
       this.getConnectionsRequestsSize();
+      this.getGamificationPoints()
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Add ; here

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue (hunk 15 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/TotalPoints.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue (hunk 16 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

**Pull request merged**